### ISSUE TITLE
Flutter and Symbol/Zebra MC18 Android

### ISSUE THREAD

**Issue by mark-revision** [ZebraDevs/DataWedge-Flutter-Demo#7]
Hi!

We are experiencing major problems on the SymbolZebra MC18 in combination with Flutter. We've built apps using DataWedge on MC18 and PS20 devices, but performance on MC18 is terrible, making the device unusable. User presses a button and the app freezes for 2 seconds... We know this is a rather old device with a limited CPU/GPU, but performance is ok, when these lag spikes are gone. We have already filed a issue for it for the Flutter Team (https://github.com/flutter/flutter/issues/71315), but we're wondering...

- if other people are experiencing the same issue and maybe have a workaround for it
- if Zebra maybe has a fix for this (seems to be GPU related, see ticket above)

We haven't encountered this bug with native Android development or Xamarin development.

Best regards,

Mark

**Comment by darryncampbell** [comment]
Looking at the issue you filed with the Flutter team, I would not be able to provide any further comment as it looks comprehensive to me and you seem to have isolated the issue to Flutter's handling of the PowerVR SGX540 chip so I will need to defer to any response you get on that thread.

I can confirm that we have not seen this reported elsewhere but that is likely because Flutter is not widely adopted amongst our developer community.

Let's keep this issue open until your flutter ticket gets some resolution.

**Comment by xbr786** [comment]
Re-Vision also created and Support case and we have an Engineering query referring the same within Zebra  under SPR 41631

**Comment by darryncampbell** [comment]
Understood.  Please note from the readme: "Please be aware that this application / sample is provided as-is for demonstration purposes without any guarantee of support"

**Comment by mark-revision** [comment]
Hi @darryncampbell and @xbr786 

Just a small report on this issue. We have also reported this issue to Google together with a reproducible test that identified the issue. After a meeting with some Google engineers, sharing tracelogs and sending an MC18 to Google in March/April, Google identified that "during SkCanvas::flush occurred during calls to glTexSubImage2D using the GL_RED texture format". This issue (GPU Driver bug) could be worked around by no longer using the GL_RED texture format on devices with the PowerVR SGX 540 GPU.  At the end of April this fix landed on Flutter's Master channel, later it was merged into the Beta channel and since Flutter 2.5 that was released September 7th, this fix is now also in Flutter's Stable channel. So Flutter apps are now running again fine on MC18s and this issue on the support case SPR 41631 can be closed. Big thanks to Google's Flutter team for pocking this up and fixing this issue.

Best regards,

Mark
